cmd/mooshy/mooshy.go

```diff
@@ -174,31 +174,6 @@ func latestMoosh(token string) (string, error) {
 	return "", errors.New("not found")
 }
 
-func pollMagicNumber(conn net.Conn, port string, addr string, connected chan struct{}) error {
```

Comment:
Shouldn't the function `runShell` (and `NewSSHRunner` if it's not intentionally exported) than also be included in its/their calling function (i.e. `main`)? Both are only called once in the code.

Reply:
Yes. We need to be careful with the `defer` statements though.
They do not execute after a `log.Fatal` at all.
One way around this would be extracting most of the `main` contents into a separate function, which returns an error.

Reply:
Okay. I think that would be something for a separate issue/PR.